src/zkevm_specs/evm/instruction.py

```diff
@@ -528,6 +544,27 @@ def sub_balance_with_reversion(
         self.constrain_zero(carry)
         return balance, balance_prev
 
+    def account_storage_read(self, account_address: int, storage_key: int) -> Tuple[FQ, FQ, FQ, FQ]:
+        row = self.rw_lookup(RW.Read, RWTableTag.AccountStorage, [account_address, storage_key, 0])
+        return row[-4], row[-3], row[-2], row[-1]
+
+    def account_storage_write_with_reversion(
+        self,
+        account_address: int,
+        storage_key: int,
+        is_persistent: bool,
+        rw_counter_end_of_reversion: int,
+        state_write_counter: Optional[int] = None,
+    ) -> Tuple[FQ, FQ, FQ, FQ]:
+        row = self.state_write_with_reversion(
+            RWTableTag.AccountStorage,
+            [account_address, storage_key],
+            is_persistent,
+            rw_counter_end_of_reversion,
+            state_write_counter,
+        )
+        return row[-4], row[-3], row[-2], row[-1]
+
```

Comment:
I think it makes more sense to treat `tx_id` as input (as mentioned above), and we don't need to return it.

```suggestion
    def account_storage_write_with_reversion(
        self,
        account_address: int,
        storage_key: int,
        tx_id: int,
        is_persistent: bool,
        rw_counter_end_of_reversion: int,
        state_write_counter: Optional[int] = None,
    ) -> Tuple[FQ, FQ, FQ]:
        row = self.state_write_with_reversion(
            RWTableTag.AccountStorage,
            [account_address, storage_key, Tables._, Tables._, tx_id],
            is_persistent,
            rw_counter_end_of_reversion,
            state_write_counter,
        )
        return row[-4], row[-3], row[-1]

```